### ISSUE TITLE
Compiler error in `scatter`

### ISSUE THREAD

**Issue by cmrfrd** [aws-neuron/aws-neuron-sdk#790]
The following XLA hlo meant to mimic the semantics of `numpy.put_along_axis` produces a compiler error via `neuronx-cc`

```
ENTRY main.44 {
  constant.1 = f32[] constant(0)
  broadcast.2 = f32[3,4]{1,0} broadcast(constant.1), dimensions={}
  iota.28 = s32[3]{0} iota(), iota_dimension=0
  reshape.29 = s32[3,1]{1,0} reshape(iota.28)
  reshape.30 = s32[3]{0} reshape(reshape.29)
  broadcast.31 = s32[3,4]{1,0} broadcast(reshape.30), dimensions={0}
  reshape.32 = s32[3,4,1]{2,1,0} reshape(broadcast.31)
  constant.9 = s32[3,4]{1,0} constant({...})
  reshape.33 = s32[3,4,1]{2,1,0} reshape(constant.9)
  concatenate.34 = s32[3,4,2]{2,1,0} concatenate(reshape.32, reshape.33), dimensions={2}
  constant.10 = f32[3,1]{1,0} constant({ {1}, {1}, {1} })
  reshape.35 = f32[3]{0} reshape(constant.10)
  broadcast.36 = f32[3,4]{1,0} broadcast(reshape.35), dimensions={0}
  scatter.37 = f32[3,4]{1,0} scatter(broadcast.2, concatenate.34, broadcast.36), update_window_dims={}, inserted_window_dims={0,1}, scatter_dims_to_operand_dims={0,1}, index_vector_dim=2, to_apply=region_0.41
  ROOT tuple.42 = (f32[3,4]{1,0}) tuple(scatter.37)
}
```
[rust_hlo_xla_put_along_axis.pb.zip](https://github.com/aws-neuron/aws-neuron-sdk/files/13394531/rust_hlo_xla_put_along_axis.pb.zip)

error:

```
[neuronxcc.driver.CommandDriver]: ***************************************************************
[neuronxcc.driver.CommandDriver]:  An Internal Compiler Error has occurred
[neuronxcc.driver.CommandDriver]: ***************************************************************
[neuronxcc.driver.CommandDriver]: 
[neuronxcc.driver.CommandDriver]: ('More than 1 par dims %s', frozenset({0, 1}))
[neuronxcc.driver.CommandDriver]: 
```

The HLO produced above was compiled via our rust framework, but semantically equivalent HLO can be reproduced using the following jax code

```python
def f():
    base = jnp.zeros((3, 4), dtype=jnp.float32)
    indices = (
        jnp.array([[0, 1, 2]], dtype=int).reshape((3, 1)),
        jnp.zeros((3, 4), dtype=int),
    )
    values = jnp.array([1, 1, 1], dtype=jnp.float32).reshape((3, 1))
    axis = 1
    result = base.at[indices].add(values)
    return result

comp = jax.xla_computation(f)()
print(comp.as_hlo_text())
```

The result of both should be:

```
 [[4. 0. 0. 0.]
 [4. 0. 0. 0.]
 [4. 0. 0. 0.]]
```

Command and compiler version:

```
neuronx-cc compile /mnt/drive1/tmp/rust_hlo_xla_put_along_axis.pb --framework=XLA --target=trn1 --model-type=transformer --auto-cast=none --output /mnt/drive1/tmp/xla_put_along_axis_8295513037111426501.neff
```

```
$ neuronx-cc --version
NeuronX Compiler version 2.11.0.34+c5231f848

Python version 3.8.10
HWM version 2.11.0.2-e34678757
NumPy version 1.24.4
```

**Comment by aws-mvaria** [comment]
Thank you, we are taking a look and will get back to you shortly.

**Comment by aws-donkrets** [comment]
Hi [cmrfrd](https://github.com/cmrfrd) - quick update... we have identified a fix for this issue and expect to have it resolved in the next SDK release.

**Comment by cmrfrd** [comment]
I have re-tested this on the newest release

```
$ neuronx-cc --version
NeuronX Compiler version 2.12.40.0+4928b2c8b
```

It compiles this time but produces incorrect output:

```
 [[4. 0. 0. 0.]
 [0. 4. 0. 0.]
 [0. 0. 4. 0.]]
 ```
 
 Can you reproduce this?

**Comment by cmrfrd** [comment]
Another reproduction of this bug via `torch-xla`

```python
import torch_xla.core.xla_model as xm
import torch
import os
def f(xla_device):
    axis = 1
    a = torch.zeros((3, 4), dtype=torch.float32, device=xla_device)
    index = torch.broadcast_to(
        torch.zeros(3, dtype=torch.int64, device=xla_device).reshape((3, 1)), (3, 4)
    )
    value = torch.broadcast_to(
        torch.ones((3, 1), dtype=torch.float32, device=xla_device), (3, 4)
    )
    return torch.scatter_add(a, axis, index, value)

print("Running f via torch cpu")
print(f("cpu"))

print("Running f via torch xla + trn")
os.environ['NEURON_FRAMEWORK_DEBUG'] = '1'
os.environ['XLA_DENSE_SCATTER_FACTOR'] = '0'
print(f(xm.xla_device()))
```

produces

```
Running f via torch cpu
tensor([[4., 0., 0., 0.],
        [4., 0., 0., 0.],
        [4., 0., 0., 0.]])
Running f via torch xla + trn
tensor([[4., 0., 0., 0.],
        [0., 4., 0., 0.],
        [0., 0., 4., 0.]], device='xla:1')
```

**Comment by aws-donkrets** [comment]
[cmrfrd](https://github.com/cmrfrd) - thx for the quick test and additional feedback. update. We are testing a potential fix for this. If successful, it should make the next SDK release.

**Comment by xanderdunn** [comment]
This appears to be fixed in `neuronx_cc-2.12.54.0+f631c2365`, thank you! :) 

We'll close this once it hits release

**Comment by xanderdunn** [comment]
This is fixed in 2.16 release, @cmrfrd or AWS, you can close it.
